modules/aggs-matrix-stats/src/main/java/org/elasticsearch/search/aggregations/support/ArrayValuesSourceAggregatorFactory.java

```diff
@@ -58,17 +59,27 @@ public Aggregator createInternal(SearchContext searchContext,
         if (valuesSources.isEmpty()) {
             return createUnmapped(searchContext, parent, metadata);
         }
-        return doCreateInternal(valuesSources, searchContext, parent, collectsFromSingleBucket, metadata);
+        return doCreateInternal(valuesSources, searchContext, parent, cardinality, metadata);
     }
 
+    /**
+     * Create the {@linkplain Aggregator} for a field that isn't mapped.
+     */
     protected abstract Aggregator createUnmapped(SearchContext searchContext,
                                                     Aggregator parent,
                                                     Map<String, Object> metadata) throws IOException;
 
+    /**
+     * Create the {@linkplain Aggregator} for a mapped field.
```

Comment:
This isn't accurate.  If the user specified a script, we might not have a mapped field at this point.  Or for that matter, if we couldn't map a field but the user provided a missing value.  It would be accurate to say we are creating the aggregator for a non-empty values source.

In fact, a large point of the values source abstraction is to not have to know at this point if we have a mapped field or not.

Reply:
Bah. I meant to revert that too after our conversation yesterday.